### PR TITLE
unlimited assets: backwards-compatible JSON encodings for account data

### PR DESCRIPTION
**Pull request by cce**
## Summary

This restores the old JSON field names in the account data blob.

## Test Plan

Existing tests should pass.

### CODE REVIEW

File: idb/postgres/internal/encoding/types.go [hunk 1]
```diff
@@ -111,26 +111,32 @@ type specialAddresses struct {
...
-	AuthAddr                   crypto.Digest `codec:"b"`
-	TotalAppSchemaNumUint      uint64        `codec:"c"`
-	TotalAppSchemaNumByteSlice uint64        `codec:"d"`
-	TotalExtraAppPages         uint32        `codec:"e"`
-	TotalAssetParams           uint64        `codec:"f"`
-	TotalAssets                uint64        `codec:"g"`
-	TotalAppParams             uint64        `codec:"h"`
-	TotalAppLocalStates        uint64        `codec:"i"`
+	Status              basics.Status `codec:"onl"`
+	AuthAddr            crypto.Digest `codec:"spend"`
```
Comment (winder) [inline idb/postgres/internal/encoding/types.go:126]: There is a query that needs to be change `b` to `spend`.

**Review by winder — CHANGES_REQUESTED** [review]

File: idb/postgres/internal/encoding/types.go [hunk 2]
```diff
@@ -111,26 +111,32 @@ type specialAddresses struct {
...
+	TotalAppSchema      stateSchema   `codec:"tsch"`
+	TotalExtraAppPages  uint32        `codec:"teap"`
+	TotalAssetParams    uint64        `codec:"tasp"`
+	TotalAssets         uint64        `codec:"tas"`
+	TotalAppParams      uint64        `codec:"tapp"`
+	TotalAppLocalStates uint64        `codec:"tapl"`
 
 	baseOnlineAccountData
 }
+
```
Comment (tolikzinovyev) [inline idb/postgres/internal/encoding/types.go:136]: nit: move this up

File: idb/postgres/internal/encoding/encoding.go [hunk 3]
```diff
@@ -681,15 +681,17 @@ func TrimLcAccountData(ad ledgercore.AccountData) ledgercore.AccountData {
...
-		TotalAppSchemaNumUint:      ad.TotalAppSchema.NumUint,
-		TotalAppSchemaNumByteSlice: ad.TotalAppSchema.NumByteSlice,
-		TotalExtraAppPages:         ad.TotalExtraAppPages,
-		TotalAssetParams:           ad.TotalAssetParams,
-		TotalAssets:                ad.TotalAssets,
-		TotalAppParams:             ad.TotalAppParams,
-		TotalAppLocalStates:        ad.TotalAppLocalStates,
+		Status:   ad.Status,
+		AuthAddr: crypto.Digest(ad.AuthAddr),
+		TotalAppSchema: stateSchema{
```
[the remark this answers is not in the document]
Comment (cce) [reply idb/postgres/internal/encoding/encoding.go:686]: oh no difference! good point

File: idb/postgres/internal/encoding/types.go (hunk 1 above)
Comment (cce) [reply idb/postgres/internal/encoding/types.go:126]: terrifying, thanks

**Review by tolikzinovyev — APPROVED** [review]

File: idb/postgres/internal/encoding/types.go (hunk 1 above)
Comment (winder) [reply idb/postgres/internal/encoding/types.go]: This still isn't resolved, you need to revert this change in `postgres.go`:
```
  if len(opts.EqualToAuthAddr) > 0 {
    whereParts = append(whereParts, fmt.Sprintf("a.account_data ->> 'b' = $%d", partNumber))
    whereArgs = append(whereArgs, encoding.Base64(opts.EqualToAuthAddr))
    partNumber++
  }
```

Comment (cce) [reply idb/postgres/internal/encoding/types.go]: thanks! i left that in a stash, updated now

**Review by winder — APPROVED** [review]

**Pull request merged**